Sources/CacheAdvance/CacheReader.swift

```diff
@@ -75,16 +72,8 @@ final class CacheReader {
             return message
 
         case .emptyRead:
-            guard offsetInFileAtEndOfNewestMessage < startingOffset else {
-                // We started reading before the offset of the end of the newest message, therefore we expected a message to be read. We instead read an empty space, meaning that the file is corrupt.
-                throw CacheAdvanceError.fileCorrupted
-            }
-
-            // We know the next message is at the end of the file header. Let's seek to it.
-            try reader.seek(to: FileHeader.expectedEndOfHeaderInFile)
-
-            // We know there's a message to read now that we're at the start of the file.
-            return try nextEncodedMessage()
+            // We hit an empty read when reach EOF, so return nil to stop.
```

Comment:
Thoughts on the following comment?
```suggestion
            // An empty read means we hit the EOF. It is the responsibility of the calling code to validate this assumption.
```